executors/clipseg/executor.py

```diff
@@ -0,0 +1,214 @@
+import enum
+import os
+import sys
+import shutil
+import time
+
+from PIL import Image, ImageOps
+from io import BytesIO
+from pathlib import Path
+from typing import Dict
+from urllib.request import urlopen
+
+import cv2
+import numpy as np
+import torch
+
+from jina import Executor, DocumentArray, Document, requests
+from torchvision import transforms
+
+
+class THRESHOLDING_METHODS(str, enum.Enum):
+    NONE = 'none' # Do not threshold
+    BINARY = 'binary'
+    ADAPTIVE_MEAN = 'adaptive_mean'
+    ADAPTIVE_GAUSSIAN = 'adaptive_gaussian'
+
+
+DALLE_CACHE_DOCKER_DIR = '/home/dalle/.cache'
+THRESHOLD_ADAPTIVE_DEFAULT_BLOCK_SIZE = 11
+THRESHOLD_ADAPTIVE_DEFAULT_C = 2.
+THRESHOLD_BINARY_DEFAULT_STRENGTH_VALUE = 85
+SELF_DIR = os.path.dirname(os.path.abspath(__file__))
+WEIGHTS_LOC = 'https://owncloud.gwdg.de/index.php/s/ioHbRzFx6th32hn/download'
+WEIGHT_FOLDER_NAME = 'clipseg_weights'
+WEIGHT_ZIP_FILE_NAME = 'clipseg_weights.zip'
+
+TRANSFORM_FOR_SEGMENTATION = transforms.Compose([
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
+    transforms.Resize((512, 512)),
+])
+
+
+def document_to_pil(doc):
+    uri_data = urlopen(doc.uri)
+    return Image.open(BytesIO(uri_data.read()))
+
+
+class ClipSegmentation(Executor):
+    clipseg_path = None
+    model = None
+
+    def __init__(self, clipseg_path: str, **kwargs):
+        super().__init__(**kwargs)
+        self.clipseg_path = clipseg_path
+        sys.path.append(str(Path(clipseg_path).resolve().absolute()))
+
+        weights_path = Path('/')
+        if Path(DALLE_CACHE_DOCKER_DIR).is_dir():
+            weights_path = Path(DALLE_CACHE_DOCKER_DIR) / WEIGHT_ZIP_FILE_NAME
+        else:
+            # Assume we're working locally, use local home.
+            weights_path = Path.home() / WEIGHT_ZIP_FILE_NAME
+
+        if not weights_path.is_file():
+            response = urlopen(WEIGHTS_LOC)
+            weights_bytes = response.read()
+            with open(weights_path, 'wb') as w_f:
+                w_f.write(weights_bytes)
+
+        shutil.unpack_archive(weights_path, Path(clipseg_path).resolve())
+
+        from models.clipseg import CLIPDensePredT
```

Comment:
can we move this import to the top ?

Reply:
It's only provided by `sys.path.append(str(Path(clipseg_path).resolve().absolute()))`, and `clipseg_path` is configuration level.